src/target/target.c

```diff
@@ -589,8 +589,14 @@ int target_halt(struct target *target)
 	int retval;
 	/* We can't poll until after examine */
 	if (!target_was_examined(target)) {
-		LOG_ERROR("Target not examined yet");
-		return ERROR_FAIL;
+		/* Try to examine the target right now, in case the target we're
+		 * talking to didn't examine correctly during `init`. */
+		LOG_TARGET_INFO(target, "Try to examine unexamined target in target_halt().");
+		target_examine();
+		if (!target_was_examined(target)) {
+			LOG_ERROR("Target not examined yet");
+			return ERROR_FAIL;
+		}
```

Comment:
I'd recommend to not mention concrete functions (`target_halt()`) since the messages will be read by users that do not know OpenOCD internals.

For instance:

```suggestion
		LOG_TARGET_INFO(target, "Trying to examine unexamined target before halt attempt.");
		target_examine();
		if (!target_was_examined(target)) {
			LOG_TARGET_ERROR(target, "Re-examination before halt failed. Target not examined yet.");
			return ERROR_FAIL;
		}
```